extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -488,8 +539,14 @@ export class AmpA4A extends AMP.BaseElement {
                   if (isValid) {
                     return creative;
                   }
-                  return Promise.reject(
-                      'Key failed to validate creative\'s signature.');
+                  // Only report if signature is expected to match, given that
+                  // multiple key providers could have been specified.
+                  if (verifyHashVersion(signature, keyInfo)) {
```

Comment:
What happens if the key hash in the signature doesn't match any available key? That would be an error case, but I don't see how it gets reported.